src/inference/src/dev/core_impl.hpp

```diff
@@ -95,13 +95,34 @@ class CoreImpl : public InferenceEngine::ICore, public std::enable_shared_from_t
             std::shared_ptr<InferenceEngine::ICacheManager> _cacheManager;
         };
 
-        bool flag_allow_auto_batching = true;
+        // Set core properties with input config map, it will update core properties data.
+        void set_core_config(ov::AnyMap& config);
 
-        void set_and_update(ov::AnyMap& config);
+        // Return whether this config belongs to core config
+        bool is_core_config(const std::string& config_name) const;
 
-        void set_cache_dir_for_device(const std::string& dir, const std::string& name);
+        // Check whether this core config is supported by plugin, and the coresponding propertie will be removed
+        // from plugin.set_property().
+        bool core_config_supported(const ov::Plugin& plugin, const std::string& config_name) const;
+
+        // Intercept core config that will be set to plugin by calling plugin.set_property().
+        // Put it into global config map if this plugin supported
+        void intercept_core_config(ov::Plugin& plugin, ov::AnyMap& config);
+
+        // Clean up global config for input config.
+        void cleanup_core_config(ov::Plugin& plugin, ov::AnyMap& config) const;
+
+        // Update input config with the help of core_plugins_properties:
+        //  1. Remove the core config that this plugin doesn't support
+        //  2. Don't overwritten this config if they have been in input config
+        //  3. Add default core config into input config if it doesn't have.
+        template <typename T>
+        void update_config(ov::Plugin& plugin, std::map<std::string, T>& config) const;
```

Comment:
Why do we need to introduce so much helpers which do some black magic with original config?

Reply:
They are used to put core properties handling in the same place, which will be easy to maintained.